src/components/hv-picker-field/index.js

```diff
@@ -0,0 +1,247 @@
+   * Uses styles defined on the <picker-field> element for the modal and buttons.
+   */
+  renderPickerModal = () => {
+    const { element, stylesheets, options } = this.props;
+    const modalStyle = createStyleProp(element, stylesheets, {
+      ...options,
+      styleAttr: 'modal-style',
+    });
+    const cancelTextStyle = createStyleProp(element, stylesheets, {
+      ...options,
+      pressed: this.state.cancelPressed,
+      styleAttr: 'modal-text-style',
+    });
+    const saveTextStyle = createStyleProp(element, stylesheets, {
+      ...options,
+      pressed: this.state.savePressed,
+      styleAttr: 'modal-text-style',
+    });
+    const cancelLabel = element.getAttribute('cancel-label') || 'Cancel';
+    const saveLabel = element.getAttribute('save-label') || 'Save';
+
+    return (
+      <Modal
+        animationType="slide"
+        transparent
+        visible={this.state.showPicker}
+        onRequestClose={this.cancelPickerModal}
+      >
+        <View
+          style={{
+            flex: 1,
+            justifyContent: 'flex-end',
+          }}
+        >
+          <View style={modalStyle}>
+            <View
+              style={{
+                flexDirection: 'row',
+                justifyContent: 'space-between',
+              }}
+            >
+              <TouchableWithoutFeedback
+                onPressIn={() => this.setState({ cancelPressed: true })}
+                onPressOut={() => this.setState({ cancelPressed: false })}
+                onPress={this.cancelPickerModal}
+              >
+                <Text style={cancelTextStyle}>{cancelLabel}</Text>
+              </TouchableWithoutFeedback>
+              <TouchableWithoutFeedback
+                onPressIn={() => this.setState({ savePressed: true })}
+                onPressOut={() => this.setState({ savePressed: false })}
+                onPress={this.savePickerModal}
+              >
+                <Text style={saveTextStyle}>{saveLabel}</Text>
+              </TouchableWithoutFeedback>
+            </View>
+            {this.renderPicker()}
+          </View>
+        </View>
+      </Modal>
+    );
+  };
+
+  render() {
+    const { element, stylesheets, options } = this.props;
+    if (element.getAttribute('hide') === 'true') {
+      return null;
+    }
+
+    const focused = this.state.showPicker;
+    const pressed = this.state.fieldPressed;
+    const props = createProps(element, stylesheets, {
+      ...options,
+      focused,
+      pressed,
+      styleAttr: 'field-style',
+    });
+    const fieldTextStyle = createStyleProp(element, stylesheets, {
+      ...options,
+      focused,
+      pressed,
+      styleAttr: 'field-text-style',
+    });
+    const value = element.getAttribute('value');
+    const placeholderTextColor = element.getAttribute('placeholderTextColor');
+
+    if (!value && placeholderTextColor) {
+      fieldTextStyle.push({ color: placeholderTextColor });
```

Comment:
This could be moved where `fieldTextStyle` is initialized to avoid mutation. Usually not a good idea to mutate objects.

Reply:
@divyanshu013 I don't agree in this case, mutation is the cleanest way to express the desired outcome, and there are no side-effects outside of the function. If there's a way to express the same thing without mutation, while keeping the code easy-to-read, I'm open to it.